User gets a bad request error instead of the error message

Environment: test

If a user with an older version tries to trade with the coordinator the request is rejected with a bad request.

Instead of the http error the error message should be shown.

More specifically the user should see that a new version is available if the request gets rejected due to the version check.

![image](https://github.com/get10101/10101/assets/382048/d3731fd4-4f2e-4a64-b019-d5997280e8b5)

The relevant log on the coordinator

```
2024-03-25 14:21:05 ERROR post_order: coordinator::orderbook::routes: error=BadRequest("Please upgrade to the latest version: 2.0.6")
```

The relevant logs on the app

```
flutter: [ERROR] TIME: 2024-03-25T14:21:05.583718Z r: Failed to post new order: HTTP status client error (400 Bad Request) for url (http://127.0.0.1:8000/api/orderbook/orders) order_id: 8aa7b627-79c3-4571-90b1-a4e8dcfefe62
flutter: [ERROR] TIME: 2024-03-25T14:21:05.584267Z r: Executed price not known in state Initial
flutter: [INFO] TIME: 2024-03-25T14:21:05.584896Z r: Updated order state new_state: Failed,order_id: 8aa7b627-79c3-4571-90b1-a4e8dcfefe62
flutter: [ERROR] TIME: 2024-03-25T14:21:05.632909Z Failed to submit order: FfiException(RESULT_ERROR, Failed to post order to orderbook: HTTP status client error (400 Bad Request) for url (http://127.0.0.1:8000/api/orderbook/orders), null)
```

--- a/mobile/native/src/trade/order/orderbook_client.rs
+++ b/mobile/native/src/trade/order/orderbook_client.rs
@@ -1,5 +1,6 @@
 use crate::commons::reqwest_client;
 use crate::ln_dlc::get_node_key;
+use anyhow::bail;
 use anyhow::Result;
 use commons::ChannelOpeningParams;
 use commons::NewOrder;
@@ -34,8 +35,11 @@ impl OrderbookClient {
 
         let response = client.post(url).json(&new_order_request).send().await?;
 
-        response.error_for_status()?;
-
-        Ok(())
+        if response.status().as_u16() == 200 {
+            Ok(())
+        } else {
+            let error = response.text().await?;
+            bail!("Could not create new order: {error}")
+        }
     }
 }

